pyopia/instrument/silcam.py

```diff
@@ -23,41 +25,63 @@ def timestamp_from_filename(filename):
     return timestamp
 
 
+@dataclass
+class LoadData:
+    timestamp: Any
+    img: Any
+
+
 class SilCamLoad():
     '''PyOpia pipline-compatible class for loading a single silcam image
 
+    Requires pipeline data: :class:`LoadData`
+
     Parameters
     ----------
     filename : string
         silcam filename (.silc)
 
     Returns
     -------
-    timestamp : timestamp
-        timestamp from timestamp_from_filename()
-    img : np.array
-        raw silcam image
+        data.img : :class:`LoadData`
+        data.timestamp : :class:`LoadData`
     '''
 
     def __init__(self, filename):
         self.filename = filename
 
-    def __call__(self, data):
+    def __call__(self, data: LoadData) -> LoadData:
         timestamp = timestamp_from_filename(self.filename)
         img = np.load(self.filename, allow_pickle=False)
         data['timestamp'] = timestamp
```

Comment:
Here you can create the LoadData by doing `data = LoadData(timestamp=timestamp, img=img)`

Reply:
doing this results in the output of this removing the other data that is not used in this function (in this case the classifier object)